pytorch/pytorch_ddp_1machine.py

```diff
@@ -0,0 +1,240 @@
+"""
+Optuna example that optimizes multi-layer perceptrons using PyTorch distributed.
+
+In this example, we optimize the validation accuracy of fashion product recognition using
+PyTorch distributed data parallel and FashionMNIST.
+This example is dedicated to a use case of using one machine with multiple GPUs
+where no MPI is needed.
+We optimize the neural network architecture as well as the optimizer configuration.
+As it is too time consuming to use the whole FashionMNIST dataset, we
+here use a small subset of it.
+
+You can execute this example with a command as follows.
+Device ids such GPU ids can be specified with --device_ids argument:
+    $ python pytorch_ddp_1machine.py --device_ids 1 2
+Otherwise, CPU will be used as default.
+
+Please note that this example only works with optuna >= 3.1.0.
+If you wish to use optuna < 3.1.0, you would need to pass
+`device=device_id` in TorchDistributedTrial.
+"""
+import argparse
+from functools import partial
+import os
+
+import optuna
+from optuna.trial import TrialState
+import torch
+import torch.distributed as dist
+import torch.multiprocessing as mp
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.nn.parallel import DistributedDataParallel as DDP
+import torch.optim as optim
+import torch.utils.data
+from torchvision import datasets
+from torchvision import transforms
+
+
+# Arbitrary GPUs can be selected
```

Comment:
```suggestion
```

Reply:
Thank you for the suggestion. I changed the code accordingly.